### ISSUE TITLE
The "geben" recipe file still points to the google code SVN trunk

### ISSUE THREAD

**Issue by ahungry** [melpa/melpa#3823]
It looks like Google plans to keep it up through 2016, but come 2017 this will mean it is unavailable.

I found an existing fork on github of geben, and have a discussion going with the fork owner to see if they want to have their fork submitted to MELPA for the new geben source (I also have an email out to the original author, last seen active in 2010) to find out if the project is abandoned by them.

The current "geben" package has a serious flaw with emacs 24+ that renders the with-open-network function unusable (breaking almost all networking features in emacs if geben is `require`'d).

The fork on github already has a patch that fixes this issue.  If the fork owner is not interested in his repository being published as the source, I also created a fork of it, and would be happy to take over.

**Comment by ahungry** [comment]
there are a few other package recipes still pointed here, that will surely break when the hosting for them goes down - should the authors of each package be contacted, or should they just be forked/rehosted somewhere else?

```
./confluence: :url "http://confluence-el.googlecode.com/svn/trunk/"
./eredis:(eredis :fetcher svn :url "http://eredis.googlecode.com/svn/trunk/")
./furl:(furl :fetcher hg :url "https://code.google.com/p/furl-el/")
./geben:(geben :fetcher svn :url "http://geben-on-emacs.googlecode.com/svn/trunk/")
./google-c-style: :url "http://google-styleguide.googlecode.com/svn/trunk/"
./mozc: :url "http://mozc.googlecode.com/svn/trunk/src/unix/emacs/")
./nav:(nav :url "https://code.google.com/p/emacs-nav/" :fetcher hg :files ("ack*" "nav.el"))
./protobuf-mode:               :url "http://protobuf.googlecode.com/svn/trunk/editors/"
./scheme-complete: :url "http://code.google.com/p/scheme-complete/")
./tmmofl: :url "https://code.google.com/p/phillord-emacs-packages/"
./wide-column: :url "https://code.google.com/p/phillord-emacs-packages/"
```

**Comment by purcell** [comment]
> The fork on github already has a patch that fixes this issue. If the fork owner is not interested in his repository being published as the source, I also created a fork of it, and would be happy to take over.

Would you mind also contacting the fork owner? Happy to change the recipe to point to any fork which comes with a committed maintainer. :-)

Re. the other recipes, I think we'll probably just wait until they break rather than actively chase up all the maintainers.

**Comment by ahungry** [comment]
@purcell he followed up and would like me to go ahead and handle it (https://github.com/itorres/geben-on-emacs/issues/17#issuecomment-215570902), so let me know if you'd like me to submit a PR which updates the geben recipe to my repository (https://github.com/ahungry/geben) or if you'd like to.

Thanks!

**Comment by purcell** [comment]
Yup, that would be great - thank you!

**Comment by purcell** [comment]
I'd encourage you to remove the backwards-compatibility shim code for old emacsen, and maybe even replace the use of `cl` with its successor `cl-lib` -- probably just a case of adding `;; Package-Requires: ((cl-lib "0.5"))` and prefixing all the symbols from `cl` with `cl-`.

**Comment by ahungry** [comment]
Sounds good, I'll go through the geben code sometime soon and clean it up, then submit an updated recipe for MELPA that points to the new repo.

**Comment by ahungry** [comment]
Alright, it has been updated as suggested (if you have time, let me know if I missed any of the old cl symbols).

https://github.com/ahungry/geben

I'm going to go ahead and make a MELPA pull request for the updated recipe.

**Comment by purcell** [comment]
> Alright, it has been updated as suggested (if you have time, let me know if I missed any of the old cl symbols).

The byte compiler should tell you that, actually! (Or flycheck, if you have that enabled.)

**Comment by purcell** [comment]
Fixed by #3829
